### PR TITLE
Add EnvironmentVariables to PodOptions

### PR DESCRIPTION
**Pull request by ankitjain235**
## Change Overview

This PR adds EnvironmentVariables of type []v1.Env and SecretMounts of type map[string]string to be passed to PodOptions. SecretMounts is a map of secretTypeKeyName and secret name. For e.g., {"location": "test-location-secret"}

This was added so that location and secret information can be provided to the pod running kopia repository create/connect command.

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Issues 

- fixes #issue-number

## Test Plan

- [ ] :muscle: Manual
- [x] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

**Comment by viveksinghggits** [comment]
does this effect the kanister functions, that support podOverride, somehow?

**Review by PrasadG193 — APPROVED** [review]
lgtm

**Comment by PrasadG193** [comment]
@ankitjain235 possible to add unit test for this?

File: pkg/kube/utils.go [hunk 1]
```diff
@@ -154,3 +161,26 @@ func IsNodeReady(node *v1.Node) bool {
 	}
 	return false
 }
+
+func createSecretMountSpec(secretMount map[string]string, volMounts []v1.VolumeMount, vols []v1.Volume) ([]v1.VolumeMount, []v1.Volume, error) {
```
Comment (akankshakumari393) [inline pkg/kube/utils.go:165]: Add a description for this function.

**Review by shlokc9 — APPROVED** [review]
Overall LGTM 👍🏼 
But, the branch has some merge conflict PTAL

File: pkg/kube/pod.go [hunk 2]
```diff
@@ -109,6 +110,7 @@ func GetPodObjectFromPodOptions(cli kubernetes.Interface, opts *PodOptions) (*v1
 				ImagePullPolicy: v1.PullPolicy(v1.PullIfNotPresent),
 				VolumeMounts:    volumeMounts,
 				Resources:       opts.Resources,
+				Env:             opts.EnvironmentVariables,
```
Comment (pavannd1) [inline pkg/kube/pod.go:113]: @ankitjain235 Should we check for empty/nil before setting?

**Pull request merged**